chassis/core/src/main/java/com/griddynamics/jagger/JaggerLauncher.java

```diff
@@ -166,42 +161,26 @@ public void run() {
                 if (isStandByMode) {
                     log.info("Starting Master in stand by mode...");
                     while (isStandByMode) {
-                       environmentProperties.setProperty(TEST_CONFIG_NAME_PROP, getTestConfigName());
-                       XmlWebApplicationContext context =
-                               (XmlWebApplicationContext) loadContext(directory, MASTER_CONFIGURATION, environmentProperties);
-                       initCoordinator(context);
-                       Master master = (Master) context.getBean("master");
-                       master.run();
-                       
-                       context.destroy();
+                        environmentProperties.setProperty(TEST_CONFIG_NAME_PROP, getTestConfigName());
+                        doLaunchMaster(directory);
                     }
                 } else {
                     log.info("Starting Master in right away mode...");
-                    WebApplicationContext context = loadContext(directory, MASTER_CONFIGURATION, environmentProperties);
-                    Server server = new Server(getPortFrom("master.rest.http.port", 9090));
-                    try {
-                        server.setHandler(getServletContextHandler(context));
-                        server.start();
-    
-                        initCoordinator(context);
-                        Master master = (Master) context.getBean("master");
-                        master.run();
-                    } catch (Exception e) {
-                        log.error("Error during embedded Jetty handling.", e);
-                        throw new RuntimeException(e);
-                    } finally {
-                        try {
-                            server.stop();
-                        } catch (Exception e) {
-                            throw new RuntimeException(e);
-                        }
-                    }
+                    doLaunchMaster(directory);
                 }
             }
         };
         builder.addMainTask(masterTask);
     }
     
+    private static void doLaunchMaster(final URL directory) {
+        AbstractXmlApplicationContext context = loadContext(directory, MASTER_CONFIGURATION, environmentProperties);
+        initCoordinator(context);
+        Master master = (Master) context.getBean("master");
```

Comment:
As I understand it is not an IoC way. But I'm not sure that we can avoid this initialization here.